lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -435,73 +518,65 @@ def write(chunk)
+
           if @use_grpc
-            entry.metadata.timestamp = Google::Protobuf::Timestamp.new(
-              seconds: ts_secs,
-              nanos: ts_nanos
+            entry = Google::Logging::V2::LogEntry.new(
+              labels: entry_common_labels,
+              resource: Google::Api::MonitoredResource.new(
+                type: entry_resource.type,
+                labels: entry_resource.labels.to_h
+              ),
+              severity: grpc_severity(severity),
+              timestamp: Google::Protobuf::Timestamp.new(
+                seconds: ts_secs,
+                nanos: ts_nanos
+              )
             )
-
-            entry.metadata.severity =
-              grpc_severity(compute_severity(record, entry))
-
-            set_http_request_grpc(record, entry) # FIXME
+            set_http_request_grpc(record, entry)
+            set_payload_grpc(entry_resource.type, record, entry, is_json)
           else
-            entry.metadata.timestamp = {
-              seconds: ts_secs,
-              nanos: ts_nanos
-            }
-
-            entry.metadata.severity =
-              compute_severity(record, entry)
-
+            # Remove the labels if we didn't populate them with anything.
+            entry_resource.labels = nil if entry_resource.labels.empty?
+            entry = Google::Apis::LoggingV2beta1::LogEntry.new(
+              labels: entry_common_labels,
+              resource: entry_resource,
+              severity: severity,
+              timestamp: {
+                seconds: ts_secs,
+                nanos: ts_nanos
+              }
+            )
             set_http_request(record, entry)
-          end
-
-          # If a field is present in the label_map, send its value as a label
-          # (mapping the field name to label name as specified in the config)
-          # and do not send that field as part of the payload.
-          if @label_map
-            @label_map.each do |field, label|
-              field_to_label(record, field, entry.metadata.labels, label)
-            end
-          end
-
-          if @service_name == CLOUDFUNCTIONS_SERVICE &&
-             @cloudfunctions_log_match &&
-             @cloudfunctions_log_match['execution_id']
-            entry.metadata.labels['execution_id'] =
-              @cloudfunctions_log_match['execution_id']
-          end
-
-          if @use_grpc
-            set_payload_grpc(record, entry, is_json)
-          else
-            set_payload(record, entry, is_json)
-            entry.metadata.labels = nil if entry.metadata.labels.empty?
+            set_payload(entry_resource.type, record, entry, is_json)
           end
 
           entries.push(entry)
         end
         # Don't send an empty request if we rejected all the entries.
         next if entries.empty?
 
-        log_name = log_name(tag, labels)
+        log_name = "projects/#{@project_id}/logs/#{log_name(
+          tag, group_resource.type, group_resource.labels)}"
```

Comment:
For the grpc path, `log_name` would be converted to be utf-8 encoded. For the non-grpc path, it won't.